x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceDetailTabs.tsx

```diff
@@ -57,7 +70,23 @@ export function ServiceDetailTabs({ serviceName, tab }: Props) {
     name: 'errors',
   };
 
-  const tabs = [transactionsTab, errorsTab];
+  const serviceMapTab = {
+    link: (
+      <ServiceMapLink serviceName={serviceName}>
+        {i18n.translate('xpack.apm.home.serviceMapTabLabel', {
+          defaultMessage: 'Service Map',
+        })}
+      </ServiceMapLink>
+    ),
+    render: () => <ServiceMap serviceName={serviceName} />,
+    name: 'service-map',
+  };
+
+  const tabs = [transactionsTab, errorsTab, serviceMapTab];
+
+  if (uiSettings.get(enableServiceOverview)) {
+    tabs.unshift(overviewTab);
+  }
```

Comment:
Since we're initializing `tabs` here anyways we can avoid an array mutation:
```
const tabs = uiSettings.get(enableServiceOverview) ? [overviewTab, transactionsTab, errorsTab, serviceMapTab] : [transactionsTab, errorsTab, serviceMapTab];
```

Reply:
I left the mutation since we're `push`ing on to the array later anyway and I thought is was clearer.